io/src/main/scala/sbt/nio/Glob.scala

```diff
@@ -60,17 +61,284 @@ private[sbt] sealed trait ToGlob extends Any {
   def toGlob: Glob
 }
 object Glob extends LowPriorityGlobOps {
-  private[sbt] class ConvertedFileFilter(val f: FileFilter) extends PathNameFilter {
+  import scala.language.dynamics
```

Comment:
I'll link to the discussion we had in https://github.com/sbt/io/pull/212.

Reply:
Commit message says:

> Globs conceptually map more or less identically to the globs present in zsh or bash.

This is a bit of a stretch because glob wildcard should support `*`, including `*abc*`, `?`, `[co]`, and `[a-z]`. Linux also allows `[!x]` to mean negation. See [POSIX 2.13. Pattern Matching Notation]( http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_13).

Reply:
Good point. A more accurate sentence would have been "Globs are intended to map to posix pattern matching notation with some extended features, such as recursive globbing with '**' that are found in shells like zsh".

Reply:
I'll just take out the `**` and `*` variables. I can make a plugin if I really care.

Reply:
I still dislike `baseDirectory.value ** AllPassFilter` and think that `baseDirectory.value / **` is better. Perhaps the solution really should be to just remove the symbolic operators entirely in sbt 2. Also, even though `baseDirectory.value / ** / *.txt` is kind of nice, it is confusing that you can't do `baseDirctory.value / ** / foo*.txt`. I guess adding special syntax isn't worth it when it's so limited.

Reply:
I just discovered this: https://docs.oracle.com/javase/7/docs/api/java/nio/file/FileSystem.html#getPathMatcher(java.lang.String). I think this will do what I want.

Reply:
I believe that the sbt globs should be very close to shell globs with this commit: https://github.com/sbt/io/pull/226/commits/880ed6efa85df2512acb2ed618e22a44447d4041

Reply:
Should be resolved by https://github.com/sbt/io/pull/226/commits/f7e9d5fba02596aac99dfa2f75a3a5052449fd6a